libclc/x86_64-unknown-linux/libspirv/shared/helpers.ll

```diff
@@ -0,0 +1,270 @@
+declare float @llvm.sqrt.f32(float %n)
+declare double @llvm.sqrt.f64(double %n)
+declare float @llvm.fabs.f32(float %n)
+declare double @llvm.fabs.f64(double %n)
+declare float @llvm.trunc.f32(float %n)
+declare double @llvm.trunc.f64(double %n)
+declare float @llvm.ceil.f32(float %n)
+declare double @llvm.ceil.f64(double %n)
+declare float @llvm.floor.f32(float %n)
+declare double @llvm.floor.f64(double %n)
+declare float @llvm.round.f32(float %n)
+declare double @llvm.round.f64(double %n)
+declare float @llvm.rint.f32(float %n)
+declare double @llvm.rint.f64(double %n)
+declare float @llvm.cos.f32(float %n)
+declare double @llvm.cos.f64(double %n)
+declare float @llvm.sin.f32(float %n)
+declare double @llvm.sin.f64(double %n)
+declare float @llvm.exp2.f32(float %n)
+declare double @llvm.exp2.f64(double %n)
+declare float @llvm.exp.f32(float %n)
+declare double @llvm.exp.f64(double %n)
+declare float @llvm.log10.f32(float %n)
+declare double @llvm.log10.f64(double %n)
+declare float @llvm.log.f32(float %n)
+declare double @llvm.log.f64(double %n)
+declare float @llvm.log2.f32(float %n)
+declare double @llvm.log2.f64(double %n)
+declare float @llvm.fma.f32(float %n1, float %n2, float %n3)
+declare double @llvm.fma.f64(double %n1, double %n2, double %n3)
+declare i32 @llvm.ctpop.i32(i32 %n)
+declare i8 @llvm.ctpop.i8(i8 %n)
```

Comment:
Would using the clang built-ins in the `.cl` files work? e.g `__builtin_popcount`, `__builtin_fma`, etc ...

* https://github.com/intel/llvm/blob/sycl/clang/include/clang/Basic/Builtins.def#L122